pyacq/core/client.py

```diff
@@ -78,13 +78,21 @@ class NodeProxy(object):
     def __init__(self, nodegroup, name):
         self.nodegroup = nodegroup
         self.name = name
-        self.inputs = { name:InputStreamProxy(self, name) for name in self.nodegroup.get_node_attr(self.name, '_input_specs').keys()}
-        self.outputs = { name:OutputStreamProxy(self, name) for name in self.nodegroup.get_node_attr(self.name, '_output_specs').keys()}
+        #~ self.inputs = { name:InputStreamProxy(self, name) for name in self.nodegroup.get_node_attr(self.name, '_input_specs').keys()}
+        #~ self.outputs = { name:OutputStreamProxy(self, name) for name in self.nodegroup.get_node_attr(self.name, '_output_specs').keys()}
         
     def __getattr__(self, name):
         return lambda *args, **kwargs: getattr(self.nodegroup, 'control_node')(self.name, name, *args, **kwargs)
 
     @property
+    def inputs(self):
+        return { name:InputStreamProxy(self, name) for name in self.get_input_names() }
```

Comment:
Do we really want new `InputStreamProxy` instances created every time this is called?